bundles_test.go

```diff
@@ -1021,10 +1157,200 @@ func Test_Bundles_GetNewestBundle(t *testing.T) {
 			},
 			ErrorMatcher: nil,
 		},
+
+		// Test 6 verifies that correct bundle with given labels is found
+		{
+			Bundles: []Bundle{
+				{
+					Changelogs:   []Changelog{},
+					Components:   []Component{},
+					Dependencies: []Dependency{},
+					Deprecated:   false,
+					Labels: map[string]string{
+						"provider": "aws",
+					},
+					Name:    "cluster-operator",
+					Time:    time.Unix(20, 15),
+					Version: "0.1.0",
+					WIP:     false,
+				},
+				{
+					Changelogs:   []Changelog{},
+					Components:   []Component{},
+					Dependencies: []Dependency{},
+					Deprecated:   false,
+					Labels: map[string]string{
+						"provider": "azure",
+					},
+					Name:    "cluster-operator",
+					Time:    time.Unix(40, 35),
+					Version: "0.1.0",
+					WIP:     false,
+				},
+				{
+					Changelogs:   []Changelog{},
+					Components:   []Component{},
+					Dependencies: []Dependency{},
+					Deprecated:   false,
+					Labels: map[string]string{
+						"provider": "kvm",
+					},
+					Name:    "cluster-operator",
+					Time:    time.Unix(10, 5),
+					Version: "0.1.0",
+					WIP:     false,
+				},
+				{
+					Changelogs:   []Changelog{},
+					Components:   []Component{},
+					Dependencies: []Dependency{},
+					Deprecated:   false,
+					Labels: map[string]string{
+						"provider": "aws",
+					},
+					Name:    "cluster-operator",
+					Time:    time.Unix(120, 15),
+					Version: "0.3.0",
+					WIP:     false,
+				},
+				{
+					Changelogs:   []Changelog{},
+					Components:   []Component{},
+					Dependencies: []Dependency{},
+					Deprecated:   false,
+					Labels: map[string]string{
+						"provider": "azure",
+					},
+					Name:    "cluster-operator",
+					Time:    time.Unix(140, 35),
+					Version: "0.2.0",
+					WIP:     false,
+				},
+				{
+					Changelogs:   []Changelog{},
+					Components:   []Component{},
+					Dependencies: []Dependency{},
+					Deprecated:   false,
+					Labels: map[string]string{
+						"provider": "kvm",
+					},
+					Name:    "cluster-operator",
+					Time:    time.Unix(110, 5),
+					Version: "0.4.0",
+					WIP:     false,
+				},
+			},
+			Labels: map[string]string{
+				"provider": "azure",
+			},
+			ExpectedBundle: Bundle{
+				Changelogs:   []Changelog{},
+				Components:   []Component{},
+				Dependencies: []Dependency{},
+				Deprecated:   false,
+				Labels: map[string]string{
+					"provider": "azure",
+				},
+				Name:    "cluster-operator",
+				Time:    time.Unix(140, 35),
+				Version: "0.2.0",
+				WIP:     false,
+			},
```

Comment:
I would have expected to have two bundles now because there are two having the azure label. Why is only the latest one relevant here?